Sources/SwiftInspectorCore/Tests/PropertyAnalzyerSpec.swift

```diff
@@ -0,0 +1,468 @@
+        it("detects the type name") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.name) == "FakeType"
+        }
+
+        it("detects the properties") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "",
+              access: .internal,
+              scope: .instance)
+          ]
+        }
+      }
+
+      context("when there is a property extension") {
+        beforeEach {
+          let content = """
+          public class FakeType { }
+
+          extension FakeType {
+            var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the type name") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.name) == "FakeType"
+        }
+
+        it("detects the properties") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "",
+              access: .internal,
+              scope: .instance)
+          ]
+        }
+      }
+
+      context("when there are multiple properites") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            public var thing: String = "Hello, World"
+            var foo: Int = 4
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the properties") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          let propSet = Set(result?.properties ?? [])
+          let expectedPropSet: Set<TypeWithPropInfo.PropInfo> = [
+            .init(name: "thing", typeAnnotation: "String", comment: "", access: .public, scope: .instance),
+            .init(name: "foo", typeAnnotation: "Int", comment: "", access: .internal, scope: .instance)
+          ]
+          expect(propSet) == expectedPropSet
+        }
+      }
+
+      context("when there is a static property") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            public static var thing: String = "Hello, World"
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the property") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties) == [
+            TypeWithPropInfo.PropInfo(
+              name: "thing",
+              typeAnnotation: "String",
+              comment: "",
+              access: .public,
+              scope: .static)
+          ]
+        }
+      }
+
+      context("when there is a private property") {
```

Comment:
Let's also add a test for wwhen these are the other way: `static private` instead of `private static`